### PR TITLE
use std::pair for checkFileAndSubtitle

### PR DESCRIPTION
**Pull request by neheb**
Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: src/file_request_handler.cc [hunk 1]
```diff
@@ -53,26 +53,26 @@ FileRequestHandler::FileRequestHandler(std::shared_ptr<ContentManager> content,
 {
 }
 
-static bool checkFileAndSubtitle(fs::path& path, const std::shared_ptr<CdsObject>& obj, const std::size_t& resId, std::string& mimeType, struct stat& statbuf, const std::string& rh)
+static std::pair<bool, struct stat> checkFileAndSubtitle(fs::path& path, const std::shared_ptr<CdsObject>& obj, const std::size_t& resId, std::string& mimeType, std::string_view rh)
```
Comment (KarlStraussberger) [inline src/file_request_handler.cc:56]: No need for string_view here. It's converting a string to a new object.

**Comment by neheb** [comment]
Hmmm this mimeType parameter looks like can be removed. The function sets it based on rh.empty().

**Comment by KarlStraussberger** [comment]
> 
> 
> Hmmm this mimeType parameter looks like can be removed. The function sets it based on rh.empty().

It's an output parameter (no const!)

**Comment by neheb** [comment]
I mean this basically:

```
--- a/src/file_request_handler.cc
+++ b/src/file_request_handler.cc
@@ -53,13 +53,12 @@ FileRequestHandler::FileRequestHandler(std::shared_ptr<ContentManager> content,
 {
 }

-static std::pair<bool, struct stat> checkFileAndSubtitle(fs::path& path, const std::shared_ptr<CdsObject>& obj, const std::size_t& resId, std::string& mimeType, const std::string& rh)
+static std::pair<bool, struct stat> checkFileAndSubtitle(fs::path& path, const std::shared_ptr<CdsObject>& obj, const std::size_t& resId, const std::string& rh)
 {
     std::pair<bool, struct stat> r;

     if (!rh.empty()) {
         auto resPath = obj->getResource(resId)->getAttribute(R_RESOURCE_FILE);
-        mimeType = MIMETYPE_TEXT;
         r.first = !resPath.empty();
         if (r.first) {
             path = resPath;
@@ -101,8 +100,8 @@ void FileRequestHandler::getInfo(const char* filename, UpnpFileInfo* info)
     auto item = std::dynamic_pointer_cast<CdsItem>(obj);

     fs::path path = item ? item->getLocation() : "";
-    std::string mimeType;
-    auto [isSrt, statbuf] = checkFileAndSubtitle(path, obj, resId, mimeType, rh);
+    std::string mimeType = !rh.empty() ? MIMETYPE_TEXT : "";
+    auto [isSrt, statbuf] = checkFileAndSubtitle(path, obj, resId, rh);

     UpnpFileInfo_set_IsReadable(info, access(path.c_str(), R_OK) == 0);


```

**Comment by KarlStraussberger** [comment]
Valid point. But maybe there is a way (or already available) to get the mime type from the resource, like `text/srt`. Could be somewhere in metacontent_handler.

**Comment by KarlStraussberger** [comment]
> 
> 
> Valid point. But maybe there is a way (or already available) to get the mime type from the resource, like `text/srt`. Could be somewhere in metacontent_handler.

It can be taken from resource attribute `R_PROTOCOLINFO`. `getMTFromProtocolInfo` sould do the work

**Comment by neheb** [comment]
Right. That gets done later in the code.

**Comment by KarlStraussberger** [comment]
The best solution is to integrate the function in the getInfo code. I created it some time ago to avoid duplicate code. The second call is gone now, so can the function

**Comment by neheb** [comment]
Done. Hope I didn't make a mistake.

**Comment by KarlStraussberger** [comment]
Couldn't see any flaw. Thank you

**Pull request merged**